examples/demo/galaxy/galaxy.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python
+import numpy as np
+
+from vispy.util.transforms import perspective
+from vispy.util import transforms
+from vispy import gloo
+from vispy import app
+from vispy import io
+
+import galaxy_specrend
+from galaxy_simulation import Galaxy
+
+VERT_SHADER = """
+#version 120
+uniform mat4  u_model;
+uniform mat4  u_view;
+uniform mat4  u_projection;
+
+//sampler that maps [0, n] -> color according to blackbody law
+uniform sampler1D u_colormap;
+//index to sample the colormap at
+attribute float a_color_index;
+
+//size of the star
+attribute float a_size;
+//type
+//type 0 - stars
+//type 1 - dust
+//type 2 - h2a objects
+//type 3 - h2a objects
+attribute float a_type;
+attribute vec2  a_position;
+//brightness of the star
+attribute float a_brightness;
+
+varying vec3 v_color;
+void main (void)
+{
+    gl_Position = u_projection * u_view * u_model * vec4(a_position, 0.0,1.0);
+
+    //find base color according to physics from our sampler
+    vec3 base_color = texture1D(u_colormap, a_color_index).rgb;
+    //scale it down according to brightness
+    v_color = base_color * a_brightness;
+
+
+    if (a_size > 2.0)
+    {
+        gl_PointSize = a_size;
+    } else {
+        gl_PointSize = 0.0;
+    }
+
+    if (a_type == 2) {
+        v_color *= vec3(2, 1, 1);
+    }
+    else if (a_type == 3) {
+        v_color = vec3(.9);
+    }
+}
+"""
+
+FRAG_SHADER = """
+#version 120
+//star texture
+uniform sampler2D u_texture;
+//predicted color from black body
+varying vec3 v_color;
+
+void main()
+{
+    //amount of intensity from the grayscale star
+    float star_tex_intensity = texture2D(u_texture, gl_PointCoord).r;
+    gl_FragColor = vec4(v_color * star_tex_intensity, 0.8);
+}
+"""
+
+galaxy = Galaxy(10000)
+galaxy.reset(13000, 4000, 0.0004, 0.90, 0.90, 0.5, 200, 300)
+# coldest and hottest temperatures of out galaxy
+t0, t1 = 200.0, 10000.0
+# total number of discrete colors between t0 and t1
+n = 1000
+dt = (t1 - t0) / n
+
+# maps [0, n) -> colors
+# generate a linear interpolation of temperatures
+# then map the temperatures to colors using black body
+# color predictions
+colors = np.zeros(n, dtype=(np.float32, 3))
+for i in range(n):
+    temperature = t0 + i * dt
+    x, y, z = galaxy_specrend.spectrum_to_xyz(galaxy_specrend.bb_spectrum,
+                                              temperature)
+    r, g, b = galaxy_specrend.xyz_to_rgb(galaxy_specrend.SMPTEsystem, x, y, z)
+    r = min((max(r, 0), 1))
+    g = min((max(g, 0), 1))
+    b = min((max(b, 0), 1))
+    colors[i] = galaxy_specrend.norm_rgb(r, g, b)
+
+
+# load the PNG that we use to blend the star with
+# to provide a circular look to each star.
+def load_galaxy_star_image():
+    raw_image = io.read_png("star-particle.png")
```

Comment:
Right now the example must be run from this directory to work -- if you put this as the following, it won't:
```
raw_image = io.read_png(op.join(op.dirname(__file__), 'star-particle.png'))
```
However, we usually put binary data in the `vispy-data` repo instead of this one in order to keep this repo clean. I've uploaded the data here:
```
https://github.com/vispy/demo-data/blob/master/galaxy/star-particle.png
```
You should be able to do something like this to get the file auto-downloaded:
```
from vispy.io import load_data_file
fname = load_data_file('galaxy/star-particle.png')
```

...also, this example directory should live under `examples/demo/gloo`, not `examples/demo`.